Makefile.defs

```diff
@@ -29,35 +29,49 @@ FETCH_BASE =	https://us-east.manta.joyent.com/Joyent_Dev/public/illumos-extra
 
 ifeq ($(STRAP),strap)
 STRAPPROTO =	$(DESTDIR)
-GCC =		$(DESTDIR)/usr/gcc/$(PRIMARY_COMPILER_VER)/bin/gcc
-GXX =		$(DESTDIR)/usr/gcc/$(PRIMARY_COMPILER_VER)/bin/g++
+GCCBIN =	$(DESTDIR)/usr/gcc/$(PRIMARY_COMPILER_VER)/bin/gcc
+GXXBIN =	$(DESTDIR)/usr/gcc/$(PRIMARY_COMPILER_VER)/bin/g++
 LIBSTDCXX_RUNPATH = usr/gcc/$(PRIMARY_COMPILER_VER)/lib
 else
 STRAPPROTO =	$(DESTDIR:proto=proto.strap)
-GCC =		$(STRAPPROTO)/usr/bin/gcc
-GXX =		$(STRAPPROTO)/usr/bin/g++
+GCCBIN =	$(STRAPPROTO)/usr/bin/gcc
+GXXBIN =	$(STRAPPROTO)/usr/bin/g++
 endif
 
-PATH =		$(STRAPPROTO)/usr/bin:/usr/bin:/usr/sbin:/sbin:/opt/local/bin
+#
+# As we ship some rather downrev software, there's a good chance it might fall
+# foul of newer GCC's undefined-behavior optimizations; skip those. As it's in
+# general painful to ensure that $(CFLAGS) gets passed down correctly, we'll
+# just set it directly as the compiler...
+#
+ifneq ($(PRIMARY_COMPILER_VER),4)
+GCCBIN += -fno-aggressive-loop-optimizations
+GXXBIN += -fno-aggressive-loop-optimizations
+endif
 
-MAKE =		gmake
-MAX_JOBS =	128
-PARALLEL =	-j$(MAX_JOBS)
-MAKE_VERBOSE =	V=1
-GCC.64 =	$(GCC) -m64
-GXX.64 =	$(GXX) -m64
+GCC.32 =	$(GCCBIN) -m32
+GXX.32 =	$(GXXBIN) -m32
+GCC.64 =	$(GCCBIN) -m64
+GXX.64 =	$(GXXBIN) -m64
 
 # The version of gcc in /opt/local/bin varies depending on the pkgsrc release
 GCC.host_prefix ?= /opt/local/bin
-GCC.host =	$(GCC.host_prefix)/gcc
-GXX.host =	$(GCC.host_prefix)/g++
-GCC.host.64 =	$(GCC.host) -m64
-GXX.host.64 =	$(GXX.host) -m64
+GCC.host.32 =	$(GCC.host_prefix)/gcc -m32
+GXX.host.32 =	$(GCC.host_prefix)/g++ -m32
+GCC.host.64 =	$(GCC.host_prefix)/gcc -m64
+GXX.host.64 =	$(GCC.host_prefix)/g++ -m64
 
 # We need gcc 4.x in order to build our copy of gcc 4.4. This is set by
 # 'configure' in smartos-live.
-GCC4.host =	$(HOST_GCC4_PREFIX)/gcc
-GXX4.host =	$(HOST_GCC4_PREFIX)/g++
+GCC4.host =	$(HOST_GCC4_PREFIX)/gcc -m32
```

Comment:
Should we comment here that this _must_ be -m32, and if it changes, be sure to also change ./gcc4/Makefile which does `GCC.host.32 = $(GCC4.host)`

Reply:
sure